lib/Task.js

```diff
@@ -15,56 +15,71 @@ const
 class Task {
   constructor(opt) {
     // Run only once
-
     /* istanbul ignore next */
     if (this.called) return
     this.called = true
 
-    // make sure the parsers object is always valid
-    opt.parsers = helpers.extend(
-      compiler.parsers,
-      opt.parsers || {}
-    )
-
-    // validate the compiler options
-    this.error = opt.compiler ? this.validate(opt.compiler, opt.parsers) : null
+    this.error = null
 
     // create a regex to figure out whether our user
     // wants to compile a single tag or some tags in a folder
     this.extRegex = new RegExp(`\\.${opt.ext || 'tag' }$`)
 
-    // If no target dir, default to source dir
-
-    if (!opt.to)
-      opt.to = this.extRegex.test(opt.from) ? path.dirname(opt.from) : opt.from
-
-    // Resolve to absolute paths
-
-    opt.from = path.resolve(opt.from)
-    opt.to = path.resolve(opt.to)
-
-    // Check if the path exsists
-    if (!sh.test('-e', opt.from)) this.error = NO_FILE_FOUND
+    // make sure the parsers object is always valid
+    opt.parsers = helpers.extend(
+      compiler.parsers,
+      opt.parsers || {}
+    )
 
-    // throw the error only in the cli
-    if (this.error) {
-      /* istanbul ignore next */
-      if (opt.isCli)
-        helpers.err(this.error)
-      else return this.error
+    if (opt.compiler) {
+      // validate the compiler options
+      const err = this.validate(opt.compiler, opt.parsers)
+      if (err) return this.handleError(err, opt.isCli)
+    } else {
+      // make sure to set always the compiler options
+      opt.compiler = {}
     }
 
-    // Determine the input/output types
+    if (!opt.stdin) {
```

Comment:
Umm, I tried to find better expression, but couldn't. Do you have any sugesstion?